android/app/src/main/java/org/cryptimeleon/incentive/app/domain/model/PromotionState.kt

```diff
@@ -0,0 +1,15 @@
+package org.cryptimeleon.incentive.app.domain.model
+
+import org.cryptimeleon.math.structures.cartesian.Vector
+import java.math.BigInteger
+
+/**
+ * Represents the current user state for a certain promotion.
+ */
+data class PromotionState(
```

Comment:
I do not really like this name, I feel like it should have User in it to be descriptive.
UserState might be lacking the word promotion, so maybe something like UserPromotionState.

Reply:
Good idea, renamed it to `UserPromotionState` as suggested